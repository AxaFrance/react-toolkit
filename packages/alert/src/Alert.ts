import { withProps, compose } from '@axa-fr/react-toolkit-core';
import AlertCore, { AlertCoreProps } from './AlertCore';
import icons from './icons';

const defaultClassModifier = 'error';

type AlertProps = Pick<AlertCoreProps, Exclude<keyof AlertCoreProps, 'iconClassName'>> & {
  icon?: string;
};

const setWithProps = ({ icon, ...otherProps }: AlertProps): AlertCoreProps => {
  const firstModifier = otherProps.classModifier.split(' ')[0];
  return {
    ...otherProps,
    iconClassName: `glyphicon glyphicon-${icon || icons[firstModifier]}`,
  };
};

const Enhanced = compose(withProps<AlertProps, AlertCoreProps>(setWithProps))(AlertCore);

Enhanced.displayName = 'Alert';
Enhanced.defaultProps = {
  classModifier: defaultClassModifier,
};

export default Enhanced;
