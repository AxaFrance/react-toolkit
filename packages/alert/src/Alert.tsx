import React from 'react';
import AlertCore, { AlertCoreProps } from './AlertCore';
import icons from './icons';

type AlertProps = Pick<
  AlertCoreProps,
  Exclude<keyof AlertCoreProps, 'iconClassName'>
> & {
  icon?: string;
};

const setWithProps = ({ icon, ...otherProps }: AlertProps): AlertCoreProps => {
  const firstModifier = otherProps.classModifier.split(' ')[0];
  return {
    ...otherProps,
    iconClassName: `glyphicon glyphicon-${icon || icons[firstModifier]}`,
  };
};

const Alert = ({ classModifier = 'error', ...props }: AlertProps) => {
  const customProps = setWithProps({ ...props, classModifier });
  return <AlertCore {...customProps} classModifier={classModifier} />;
};

export default Alert;
