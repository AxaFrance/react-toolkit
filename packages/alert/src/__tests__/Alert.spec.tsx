import React from 'react';
import { render } from '@testing-library/react';
import Alert from '../Alert';

describe('<Alert />', () => {
  it('renders an error alert with exclamation icon and title', () => {
    const { asFragment } = render(
      <Alert
        classModifier="error"
        title="Les caves et les garages situés dans le même corps de bâtiment que le logement assuré sont garantis d'office"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('renders an info alert with info-sign icon, a title and a body containing an unordered list', () => {
    const { asFragment } = render(
      <Alert
        classModifier="error"
        title="Les caves et les garages situés dans le même corps de bâtiment que le logement assuré sont garantis d'office">
        <ul>
          <li>Test1</li>
          <li>Test2</li>
        </ul>
      </Alert>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
