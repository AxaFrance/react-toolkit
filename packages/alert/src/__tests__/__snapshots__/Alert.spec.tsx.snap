// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert /> renders an error alert with exclamation icon and title 1`] = `
<DocumentFragment>
  <div
    class="af-alert af-alert--error"
    role="alert"
  >
    <div
      class="af-alert__title"
    >
      <div
        class="af-alert__title-icon"
      >
        <i
          class="glyphicon glyphicon-exclamation-sign"
        />
      </div>
      <div
        class="af-alert__title-text"
      >
        Les caves et les garages situés dans le même corps de bâtiment que le logement assuré sont garantis d'office
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Alert /> renders an info alert with info-sign icon, a title and a body containing an unordered list 1`] = `
<DocumentFragment>
  <div
    class="af-alert af-alert--error"
    role="alert"
  >
    <div
      class="af-alert__title"
    >
      <div
        class="af-alert__title-icon"
      >
        <i
          class="glyphicon glyphicon-exclamation-sign"
        />
      </div>
      <div
        class="af-alert__title-text"
      >
        Les caves et les garages situés dans le même corps de bâtiment que le logement assuré sont garantis d'office
      </div>
    </div>
    <div
      class="af-alert__content"
    >
      <div
        class="af-alert__content__left"
      />
      <div
        class="af-alert__content__right"
      >
        <ul>
          <li>
            Test1
          </li>
          <li>
            Test2
          </li>
        </ul>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
